### PR TITLE
[ENG-1483] Right Click -> New File

### PR DESCRIPTION
**Pull request by Rocky43007**
Creates an empty file called `Untitled` and adds `(1) -> (n)` for n number of files with the same `Untitled` name.

Works in both Ephemeral Folders and Spacedrive Locations.

Screenshot 2024-04-02 at 9 09 08 PM

### CODE REVIEW

**Comment by utkubakir** [comment]
Let's merge new folder and new file as a dropdown, can reference Windows as the experience. Would be nice to have Notepad too

**Comment by utkubakir** [comment]
image
Can't create files

**Comment by Rocky43007** [comment]
> Also what's the difference between file and empty file?

That should have been `Text File`, sorry.



> Can't create files

Looks like I messed up on a type check. I'm looking into why.

File: interface/app/$libraryId/Explorer/ParentContextMenu.tsx [hunk 1]
```diff
@@ -48,6 +50,16 @@ export default (props: PropsWithChildren) => {
 			rescan();
 		}
 	});
+	const createFile = useLibraryMutation(['files.createFile'], {
+		onError: (e) => {
+			toast.error({ title: 'Error creating file', body: `Error: ${e}.` });
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/ParentContextMenu.tsx:55]: Don't need Error: here as we already have it

File: interface/app/$libraryId/Explorer/ParentContextMenu.tsx [hunk 2]
```diff
@@ -48,6 +50,16 @@ export default (props: PropsWithChildren) => {
...
+			console.error(e);
+		},
+		onSuccess: (file) => {
+			toast.success({ title: `Created new file "${file}"` });
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/ParentContextMenu.tsx:59]: i18n

File: interface/app/$libraryId/Explorer/ParentContextMenu.tsx [hunk 3]
```diff
@@ -58,6 +70,16 @@ export default (props: PropsWithChildren) => {
 			rescan();
 		}
 	});
+	const createEphemeralFile = useLibraryMutation(['ephemeralFiles.createFile'], {
+		onError: (e) => {
+			toast.error({ title: 'Error creating file', body: `Error: ${e}.` });
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/ParentContextMenu.tsx:75]: Don't need Error: here as we already have it

File: interface/app/$libraryId/Explorer/ParentContextMenu.tsx [hunk 4]
```diff
@@ -58,6 +70,16 @@ export default (props: PropsWithChildren) => {
...
+			console.error(e);
+		},
+		onSuccess: (folder) => {
+			toast.success({ title: `Created new file "${folder}"` });
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/ParentContextMenu.tsx:79]: i18n

File: interface/app/$libraryId/Explorer/ParentContextMenu.tsx [hunk 5]
```diff
@@ -89,25 +111,67 @@ export default (props: PropsWithChildren) => {
...
+									createEphemeralFolder.mutate({
+										path: parent?.path,
+										name: null
+									});
+								}
+							}}
+						/>
+						<CM.Separator />
+						<CM.Item
+							label={t('new_file')}
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/ParentContextMenu.tsx:135]: new_note_file or new_text_file

File: interface/locales/en/common.json [hunk 6]
```diff
@@ -294,7 +294,10 @@
   "networking": "Networking",
   "networking_port": "Networking Port",
   "networking_port_description": "The port for Spacedrive's Peer-to-peer networking to communicate on. You should leave this disabled unless you have a restrictive firewall. Do not expose to the internet!",
-  "new_folder": "New folder",
+  "new": "New",
+  "new_folder": "Folder",
```
Comment (utkubakir) [inline interface/locales/en/common.json:298]: just folder if you are not gonna add "New" to front. (Same issue with others. i18n keys should be descriptive)

**Review by utkubakir — CHANGES_REQUESTED** [review]

**Review by utkubakir — APPROVED** [review]

**Pull request merged**